cmd/assetsvc/handler.go

```diff
@@ -336,3 +379,25 @@ func newChartVersionListResponse(c *models.Chart) apiListResponse {
 
 	return cvl
 }
+
+func decodeParam(param string, w http.ResponseWriter) string {
+	decodedParam, err := url.PathUnescape(param)
+	if err != nil {
+		log.WithError(err).Errorf("could not decode param %s", decodedParam)
+		if w != nil {
+			response.NewErrorResponse(http.StatusBadRequest, "could not decode param: "+param).Write(w)
+		}
+	}
+	return decodedParam
+}
+
+func getEncodedChartID(c *models.Chart) string {
+	return getEncodedChartIDString(c.Repo.Name, c.Name)
+}
+
+func getEncodedChartIDString(chartRepoName, chartName string) string {
+	return getChartID(chartRepoName, url.PathEscape(chartName))
+}
+func getChartID(chartRepoName, chartName string) string {
+	return fmt.Sprintf("%s/%s", chartRepoName, chartName)
```

Comment:
Out of interest, why is the `/` between the repo and chart also not encoded, assuming this is sent over the wire as well?

Reply:
In the database, IDs are being stored like that (eg. `bitnami/wordpress`), so we need to pass this `/`to the manager.
If we store now the encoded name, we would move from `jfrog/bitnami/wordpress`to `jfrog/bitnami%2Fwordpress`